### PR TITLE
feat: [Issuer Adapter] Support for present proof protocol

### LINKED ISSUE #135: [Issuer Adapter] Support for present-proof request from RP

### PR DESCRIPTION
**Pull request by rolsonquadras**
closes #135 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/vc/issuer/credentials.go [hunk 1]
```diff
@@ -89,3 +89,12 @@ func CreateDIDCommInitCredential(docJSON []byte) *verifiable.Credential {
 
 	return vc
 }
+
+// CreatePresentationSubmission creates presentation submission vp.
+func CreatePresentationSubmission() *verifiable.Presentation {
```
Comment (sudeshrshetty) [inline pkg/vc/issuer/credentials.go:94]: just curious, why issuer needs to create presentation submission? holder/wallet creates it.

File: pkg/internal/mock/presentproof/client.go [hunk 2]
```diff
@@ -4,7 +4,7 @@ Copyright SecureKey Technologies Inc. All Rights Reserved.
 SPDX-License-Identifier: Apache-2.0
 */
 
-package mockpresentproof
+package presentproof
```
Comment (llorllale) [inline pkg/internal/mock/presentproof/client.go:7]: I prefer `mockpresentproof.Client` myself.

File: pkg/vc/issuer/credentials.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/vc/issuer/credentials.go:94]: This is creating an VP. I tried to align the function name based on the sample resp described in https://github.com/trustbloc/edge-adapter/pull/123#issue-440924663 (Example Issuer response:).

File: pkg/internal/mock/presentproof/client.go (hunk 2 above)
Comment (rolsonquadras) [reply pkg/internal/mock/presentproof/client.go:7]: `mock` is already there in the path `pkg/internal/mock/presentproof`. Tried to align this with [aries mocks](https://github.com/hyperledger/aries-framework-go/tree/master/pkg/mock) to be consistent.

File: pkg/vc/issuer/credentials.go (hunk 1 above)
Comment (llorllale) [reply pkg/vc/issuer/credentials.go:94]: @sudeshrshetty issuer adapter needs to create presentation submissions for the rp adapter

File: pkg/internal/mock/presentproof/client.go (hunk 2 above)
Comment (llorllale) [reply pkg/internal/mock/presentproof/client.go:7]: @rolsonquadras 

> Tried to align this with aries mocks to be consistent.

But why? There was nothing broken.

File: pkg/vc/issuer/credentials.go (hunk 1 above)
Comment (llorllale) [reply pkg/vc/issuer/credentials.go:94]: The `request-presentation` message is coming from the RP adapter

File: pkg/internal/mock/presentproof/client.go (hunk 2 above)
Comment (rolsonquadras) [reply pkg/internal/mock/presentproof/client.go:7]: > @rolsonquadras
> 
> > Tried to align this with aries mocks to be consistent.
> 
> But why? There was nothing broken.

It stutters and better to be consistent with other projects.

Comment (llorllale) [reply pkg/internal/mock/presentproof/client.go:7]: @rolsonquadras 

> > > Tried to align this with aries mocks to be consistent.
> > 
> > 
> > But why? There was nothing broken.
> 
> It stutters and better to be consistent with other projects.

It avoids import collisions, does not stutter in code, was not broken, was already merged, and was placed there as per your request.

Most importantly, these arbitrary changes may impact PRs already open.

Comment (rolsonquadras) [reply pkg/internal/mock/presentproof/client.go:7]: > @rolsonquadras
> 
> > > > Tried to align this with aries mocks to be consistent.
> > > 
> > > 
> > > But why? There was nothing broken.
> > 
> > 
> > It stutters and better to be consistent with other projects.
> 
> It avoids import collisions, does not stutter in code, was not broken, was already merged, and was placed there as per your request.
> 
> Most importantly, these arbitrary changes may impact PRs already open.

@llorllale what do you mean by "was placed there as per your request." ?

Comment (llorllale) [reply pkg/internal/mock/presentproof/client.go:7]: https://github.com/trustbloc/edge-adapter/pull/112#discussion_r444205574

Comment (rolsonquadras) [reply pkg/internal/mock/presentproof/client.go:7]: Yeah, but I don't see I mentioned anything about package name.

Comment (rolsonquadras) [reply pkg/internal/mock/presentproof/client.go:7]: Now we have two packages without `mock` and one with `mock` prefix - https://github.com/trustbloc/edge-adapter/tree/master/pkg/internal/mock.

It seems like our individual preferences are blocking the PRs.

@sudeshrshetty @fqutishat @llorllale

Comment (llorllale) [reply pkg/internal/mock/presentproof/client.go:7]: @rolsonquadras fine

File: pkg/vc/issuer/credentials.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/vc/issuer/credentials.go]: @sudeshrshetty [Updated](https://github.com/trustbloc/edge-adapter/pull/140/files#diff-2328620458fc9d4a14b1f25ec1e0fd38R94) the function name. As discussed with @sudeshrshetty and @llorllale , the request between rp and issuer will not contain presentation definitions.

**Pull request merged**